client/src/app/shared/shared-account-avatar/account-avatar.component.ts

```diff
@@ -23,17 +23,40 @@ export class AccountAvatarComponent {
     this._title = value
   }
 
+  defaultAvatarUrl = Account.GET_DEFAULT_AVATAR_URL()
+
   private _title: string
 
   get title () {
     return this._title || $localize`${this.account.name} (account page)`
   }
 
   get class () {
-    return `avatar avatar-${this.size}`
+    return `avatar avatar-${this.size}` + (this.avatarUrl ? '' : ` initial ${this.getColorTheme()}`)
   }
 
   get avatarUrl () {
     return Account.GET_ACTOR_AVATAR_URL(this.account)
   }
+
+  get initial () {
+    return this.account?.name.slice(0, 1)
+  }
+
+  private getColorTheme () {
+    const themes = {
+      abc: 'blue',
+      def: 'green',
+      ghi: 'purple',
+      jkl: 'gray',
+      mno: 'yellow',
+      pqr: 'orange',
+      stv: 'red',
+      wxyz: 'dark-blue'
+    }
+
+    const theme = Object.keys(themes).find(chars => chars.includes(this.initial))
+
+    return themes[theme]
+  }
```

Comment:
Please use a simple hash-based colour generation function, which allows more diversity even for two names beginning by the same letter.
i.e. https://stackoverflow.com/questions/3426404/create-a-hexadecimal-colour-based-on-a-string-with-javascript

Reply:
Looks nice, but how do we ensure the contrast between text and bg is good? And can it be an ugly UI if we just generate random colors?

Reply:
You can generate colors in a restricted color space.

Reply:
I'm sorry but I don't understand how this should be implemented. Shall we define a set of color spaces and match those with a text color? It sounds tricky to me. And by using color spaces I'm not confident with that the contrast between bg and text is good enough, but maybe there's an easy way to ensure that?